core-nonfree/ide-nonfree/src/main/java/org/alice/stageide/personresource/PersonImp.java

```diff
@@ -145,6 +145,13 @@ void updateNebPerson() {
 
   @Override
   public void setSize(Dimension3 size) {
-    this.setScale(getScaleForSize(size));
+    Dimension3 newScale = getScaleForSize(size);
+
+    if (newScale.hasNegativeComponents()) {
```

Comment:
I looked at updateAdapterFromUI(), and doing the check there would probably make sense. If the aimed-for size is negative, should the operation just be cancelled (rather than shrinking the object to close-to-zero), or should the aimed-for size be modified to a spotive close-to-zero value instead (keeping the behavior currently in the PR)?

Reply:
I would say modify the target to some slightly greater than zero values. This way:
- The expected animation happens, to some extent at least.
- The proportions are maintained, while they would be lost at 0, 0, 0.